dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -353,23 +369,38 @@ URL findClassLocation(String className) throws ClassNotFoundException {
    * @param jarFilePaths absolute paths to jar files
    * @return map of jar file paths to classes defined in them
    */
-  private static ImmutableSetMultimap<Path, String> jarFilesToDefinedClasses(
+  @VisibleForTesting
+  static ImmutableSetMultimap<Path, String> jarFilesToDefinedClasses(
       List<Path> jarFilePaths) throws IOException {
     ImmutableSetMultimap.Builder<Path, String> pathToClasses = ImmutableSetMultimap.builder();
-
     for (Path jarFilePath : jarFilePaths) {
-      for (JavaClass javaClass : topLevelJavaClassesInJar(jarFilePath)) {
+      for (JavaClass javaClass : allJavaClassesInJar(jarFilePath)) {
         pathToClasses.put(jarFilePath, javaClass.getClassName());
-        // This does not take double-nested classes. As long as such classes are accessed
-        // only from the outer class, static linkage checker does not report false positives
-        // TODO(suztomo): enhance this so that it can work with double-nested classes
-        pathToClasses.putAll(jarFilePath, listInnerClassNames(javaClass));
       }
     }
     return pathToClasses.build();
   }
 
-  private static ImmutableSet<ClassInfo> listTopLevelClassesFromJar(URL jarFileUrl)
+  /**
+   * Returns mapping from class names to the absolute paths of the first jar file that defines the
+   * classes.
+   *
+   * @param jarFilePaths absolute paths to jar files
+   */
+  @VisibleForTesting
+  static ImmutableMap<String, Path> classToDefiningJarFile(List<Path> jarFilePaths)
```

Comment:
again would invertFrom work?

Reply:
I couldn't find a method that can handle the ordering (`putIfAbsent` below implements the order), or picking up the first path.

- `invertFrom` is inverting keys and values for multimap, but the input here is a list of Path.
- `jarFileToClasses.inverse()` would give the inverted multimap but the value is a set.